lib/named-locks.js

```diff
@@ -62,37 +78,48 @@ module.exports = {
      *
      * @param {string} name - The name of the lock to acquire.
      * @param {boolean} wait - Whether to wait for the lock.
+     * @param {integer} timeout - If we are waiting for the lock, how many milliseconds before we stop (null, or anything other than a positive number, means forever).
      * @param {function} callback - A callback(err, lock) function.
      */
-    _getLock(name, wait, callback) {
+    _getLock(name, wait, timeout, callback) {
```

Comment:
Thought for making this more extensible in the future: use an options object instead of separate parameters so you'd use it like:

```js
const options = {
  wait: true,
  timeout: 10,
};
_getLock('mylock', options, (err, lock) => { ... });
```